src/ucs/sys/sys.c

```diff
@@ -1565,6 +1584,30 @@ ucs_status_t ucs_sys_get_memlock_rlimit(size_t *rlimit_value)
     *rlimit_value = (limit_info.rlim_cur == RLIM_INFINITY) ?
                             SIZE_MAX :
                             limit_info.rlim_cur;
+
+    return UCS_OK;
+}
+
+ucs_status_t ucs_sys_get_effective_memlock_rlimit(size_t *rlimit_value)
+{
+    int lock_priv;
+
+    /* get system wide rlimit value */
+    ucs_status_t status = ucs_sys_get_memlock_rlimit(rlimit_value);
+    if(status != UCS_OK) {
+        return status;
+    }
+
+    status = ucs_sys_get_memlock_privilege(&lock_priv);
+    if(status != UCS_OK) {
+        return status;
+    }
+
+    /* overwrite system wide value for priviledged users */
+    if(lock_priv) {
```

Comment:
Can simplify the logic: first check if has CAP and return `SIZE_MAX`, otherwise continue.